core/appeal/service_test.go

```diff
@@ -1096,6 +1096,7 @@ func (s *ServiceTestSuite) TestCreate() {
 			Log(mock.Anything, appeal.AuditKeyBulkInsert, mock.Anything).Return(nil).Once()
 
 		actualError := h.service.Create(context.Background(), appeals)
+		time.Sleep(time.Millisecond)
 
 		s.Nil(actualError)
 		s.Equal(expectedResult, appeals)
```

Comment:
ideally we shouldn't delay for assertion for `actualError` and `appeals` result because those values can be checked right away. Suggesting to delay after and then explicitly call `h.assertExpectations(s.T())` instead of deferring at the beginning of unit test
```suggestion
		s.Nil(actualError)
		s.Equal(expectedResult, appeals)

		time.Sleep(time.Millisecond)
		h.assertExpectations(s.T())
```